include/dd/Package.hpp

```diff
@@ -1662,6 +1679,23 @@ namespace dd {
                             return e;
                         }
                     }
+                } else {
+                    // If level is lower, then that means there is
+                    // an identity at the current level of that DD
+                    if (x.p->v > y.p->v) {
+                        e = xCopy;
+                    } else {
+                        e = yCopy;
+                    }
+                    computeTable.insert(xCopy, yCopy, {e.p, e.w});
+                    e.w = cn.mulCached(x.w, y.w);
+
+                    if (e.w.approximatelyZero()) {
+                        cn.returnToCache(e.w);
+                        return ResultEdge::zero;
+                    }
+                    std::cout << "Cached else entered \n";
+                    return e;
```

Comment:
First of all, this seems like duplicate code to me (see a few lines above). So consider combining these parts of the code.
Secondly, I believe one needs to take extra care, to do the right thing here. I'd suggest revisiting the whole logic of this `if ... else` construct here, as there can be no case where `x.p->v == y.p->v` and one of the nodes being an identity (since identities are removed entirely as part of this PR).

One possible way forward could be to just remove the `makeIdent` function, the `isIdentity` method of the Node class, as well as the `identity` flag itself from that class and see in which places the code breaks at the moment. Then, fix the corresponding places by either removing the code if it no longer applies, or adjust accordingly for the new DD structure.